src/ufuncs.c

```diff
@@ -887,23 +887,39 @@ static void* frechet_distance_data[1] = {GEOSFrechetDistanceWrapped_r};
 #endif
 /* Project and ProjectNormalize don't return error codes. wrap them. */
 static int GEOSProjectWrapped_r(void* context, void* a, void* b, double* c) {
-  /* Handle empty points (they give segfaults) */
-  if (GEOSisEmpty_r(context, b)) {
+  /* Handle empty points (they give segfaults (for b) or give exception (for a)) */
+  if (GEOSisEmpty_r(context, a) | GEOSisEmpty_r(context, b)) {
```

Comment:
Because we now properly catch and raise the GEOS exception, I needed to add this check for `a` being empty to keep the current behaviour, as GEOS itself raises an error (without any message) for empty linestrings.